packages/aila/src/lib/agents/interact.ts

```diff
@@ -147,9 +207,104 @@ export async function interact({
         break;
       case "add":
       case "replace": {
+        // handle asyncFunction agents
+        if (agentDefinition.type === "asyncFunction") {
+          if (agentDefinition.name) {
+            if (agentDefinition.name === "mathsStarterQuiz") {
+              invariant(
+                customAgents.mathsStarterQuiz,
+                "Custom agent for maths starter quiz is required",
+              );
+
+              const quiz = await customAgents.mathsStarterQuiz({
+                document,
+              });
+
+              document = handleSectionGenerated({
+                sectionKey,
+                actionType,
+                value: quiz,
+                document,
+                onUpdate,
+              });
+
+              break;
+            } else if (agentDefinition.name === "mathsExitQuiz") {
+              invariant(
+                customAgents.mathsExitQuiz,
+                "Custom agent for maths exit quiz is required",
+              );
+
+              const quiz = await customAgents.mathsExitQuiz({
+                document,
+              });
+
+              document = handleSectionGenerated({
+                sectionKey,
+                actionType,
+                value: quiz,
+                document,
+                onUpdate,
+              });
+
+              break;
+            } else if (agentDefinition.name === "basedOn") {
+              const userMessage = messageHistory.findLast(
+                (m) => m.role === "user",
+              );
+              const userBasedOnSelection = Number(userMessage?.content?.trim());
```

Comment:
Currently this will only work if someone enters "4" or "1" etc ... not if they enter the title of the lesson.